pkg/operator/deploy/helpers.go

```diff
@@ -45,47 +46,171 @@ func decodeYAMLManifestToObject(path string, resource interface{}) error {
+	default:
+		return "", fmt.Errorf("Failed to set $DEPLOY_PLATFORM or --platform flag to a valid value. Supported platforms: [upstream, openshift, ocp-testing]")
+	}
+
+	dirStat, err = os.Stat(ansibleOperatorManifestDir)
+	if os.IsNotExist(err) {
+		return "", fmt.Errorf("Failed to stat the %s deploy platform directory '%s': %v", platform, ansibleOperatorManifestDir, err)
+	}
+	if !dirStat.IsDir() {
+		return "", fmt.Errorf("Specified %s deploy platform directory '%s' is not a directory", platform, ansibleOperatorManifestDir)
+	}
+
+	return ansibleOperatorManifestDir, nil
+}
+
+// Resource contains information about an individual resource that metering manages
+type Resource struct {
+	IsMeteringConfig bool
+	Path             string
+	Resource         interface{}
+}
+
+// InitObjectFromManifest is the driver function for building up a path to the metering-ansible-operator directory
+// from the base @manifestDir directory and initializing the metering resource/CRD objects from the YAML manifests.
+func InitObjectFromManifest(manifestDir, meteringConfigCRFile string, cfg *Config, resources *MeteringResources) ([]CRD, error) {
+	ansibleOperatorManifestDir, err := getMeteringAnsiblePath(manifestDir, cfg.Platform)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get the path to the metering-ansible-operator directory: %v", err)
+	}
+
+	pathToCRDMap := map[string]string{
+		"hiveTable":        hivetableFile,
+		"prestoTable":      prestotableFile,
+		"meteringConfig":   meteringconfigFile,
+		"report":           reportFile,
+		"reportDataSource": reportdatasourceFile,
+		"reportQuery":      reportqueryFile,
+		"storageLocation":  storagelocationFile,
+	}
+
+	crds := InitMeteringCRDSlice(ansibleOperatorManifestDir, pathToCRDMap)
+
+	for _, crd := range crds {
+		err := DecodeYAMLManifestToObject(crd.Path, crd.CRD)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to decode the YAML manifest: %v", err)
+		}
+	}
+
+	meteringResourceMap := map[string]Resource{
+		"meteringConfig": Resource{
+			Path:             meteringConfigCRFile,
+			IsMeteringConfig: true,
+			Resource:         &cfg.MeteringConfig,
+		},
+		"deployment": Resource{
+			Path:     meteringDeploymentFile,
+			Resource: &resources.Deployment,
+		},
+		"serviceAccount": Resource{
+			Path:     meteringServiceAccountFile,
+			Resource: &resources.ServiceAccount,
+		},
+		"roleBinding": Resource{
+			Path:     meteringRoleBindingFile,
+			Resource: &resources.RoleBinding,
+		},
+		"role": Resource{
+			Path:     meteringRoleFile,
+			Resource: &resources.Role,
+		},
+		"clusterRoleBinding": Resource{
+			Path:     meteringClusterRoleBindingFile,
+			Resource: &resources.ClusterRoleBinding,
+		},
+		"clusterRole": Resource{
+			Path:     meteringClusterRoleFile,
+			Resource: &resources.ClusterRole,
+		},
+	}
+
+	for name, resource := range meteringResourceMap {
+		path := filepath.Join(ansibleOperatorManifestDir, resource.Path)
+
+		if resource.IsMeteringConfig {
```

Comment:
Maybe change IsMeteringConfig to `AbsolutePath`, then it's not as much of a one-off.